_posts/2021-01-12-github-application-running-on-faasd-behind-inlets.md

```diff
@@ -0,0 +1,508 @@
+---
+layout: post
+title: How to expose your Github application running locally on faasd to the internet ?
+description: In this guide, we are going to demonstrate how we can build our Github Application and run it locally on our faasd instance then expose it to the internet using inlets to be able to Github can send events to it
+tags: inlets-pro inletsctl go github-application faasd rasperry-pi caddy tls
+author_staff_member: developer-guy
+dark_background: true
+image: /images/2021-01-12-github-application-using-go-and-inlets-pro/faasd-issue-bot.png
+date: 2021-01-12
+---
+
+## Introduction
+
+In this guide, we are going to develop a [Github application](https://docs.github.com/en/free-pro-team@latest/developers/apps) using Go, then we deploy it as a serverless function to make use of [faasd](https://github.com/openfaas/faasd) which is a lightweight & portable faas engine.We are also going to do this demo on our local environment, so we should open our function which runs on our local environment to the Internet so Github can send events to our function.In order to do that we use inlets-pro which provides secure TCP/L4 tunnels.
+
+## Pre-requisites
+
+- arkade : You can basically use arkade to install of these following tools, so we need to install arkade first.
+arkade provides a portable marketplace for downloading your favourite devops CLIs and installing helm charts, with a single command.
+
+```bash
+$ curl -sLS https://dl.get-arkade.dev | sh
+$ arkade version
+            _             _
+  __ _ _ __| | ____ _  __| | ___
+ / _` | '__| |/ / _` |/ _` |/ _ \
+| (_| | |  |   < (_| | (_| |  __/
+ \__,_|_|  |_|\_\__,_|\__,_|\___|
+
+Get Kubernetes apps the easy way
+
+Version: 0.6.35
+Git Commit: df53c4f6d9c604186b36aae7f0feb1d39940be8f
+```
+
+- inletsctl : We need inletsctl to automate the task of creating an exit-node on cloud infrastructure.
+
+```bash
+$ arkade get inletsctl
+$ inletsctl version
+ _       _      _            _   _
+(_)_ __ | | ___| |_ ___  ___| |_| |
+| | '_ \| |/ _ \ __/ __|/ __| __| |
+| | | | | |  __/ |_\__ \ (__| |_| |
+|_|_| |_|_|\___|\__|___/\___|\__|_|
+
+Version: 0.7.1
+Git Commit: 8641a9a09903f0e3b18a4ed1d388a35663a52dc4
+```
+
+- inlets-pro : We need inlets-pro to tunnel out any TCP traffic from an internal network to another network.
+
+
+```bash
+$ inletsctl download --pro
+$ inlets-pro version
+ _       _      _            _
+(_)_ __ | | ___| |_ ___   __| | _____   __
+| | '_ \| |/ _ \ __/ __| / _` |/ _ \ \ / /
+| | | | | |  __/ |_\__ \| (_| |  __/\ V /
+|_|_| |_|_|\___|\__|___(_)__,_|\___| \_/
+
+PRO edition
+
+Version: 0.7.2
+Git Commit: 5190eca94b4020ec8de47a22eb411bb17a995399
+```
+
+- multipass : We need to provision ubuntu vm's to host our faasd.In order to do that multipass cli is the best tool.
+
+```bash
+$ brew install multipass
```

Comment:
Do we want to assume that every user is a Mac user?

Multipass can be downloaded from the main website, I'd prefer if you just linked the user there instead of making them use brew and assuming their OS.

Reply:
done